datafusion/expr/src/type_coercion/functions.rs

```diff
@@ -236,6 +256,58 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_get_valid_types_variadic_equal() -> Result<()> {
+        let signature = TypeSignature::VariadicEqual(vec![DataType::Int32]);
+
+        let valid_types = get_valid_types(
+            &signature,
+            &[DataType::Int32, DataType::Int32, DataType::Int32],
+        )?;
+        assert_eq!(valid_types.len(), 1);
+        assert_eq!(
+            valid_types[0],
+            vec![DataType::Int32, DataType::Int32, DataType::Int32]
+        );
+
+        // invalid case, with int and boolean
+        let is_error = get_valid_types(
+            &signature,
+            &[DataType::Int32, DataType::Boolean, DataType::Int32],
+        )
+        .is_err();
+        assert!(is_error);
```

Comment:
Another way to test for error is doing:

```suggestion
        let is_error = get_valid_types(
            &signature,
            &[DataType::Int32, DataType::Boolean, DataType::Int32],
        )
        .uwrap_err();
```

Which is both more concise, but I think also prints out what the `OK` value is when the result is not an Error